tests/Testcontainers.Tests/Unit/Containers/Unix/Modules/Databases/CosmosDbTestcontainerTest.cs

```diff
@@ -0,0 +1,57 @@
+namespace DotNet.Testcontainers.Tests.Unit
+{
+  using System;
+  using System.Net;
+  using System.Net.Http;
+  using System.Threading.Tasks;
+  using DotNet.Testcontainers.Tests.Fixtures;
+  using Microsoft.Azure.Cosmos;
+  using Xunit;
+
+  public static class CosmosDbTestcontainerTest
+  {
+    [Collection(nameof(Testcontainers))]
+    public sealed class ConnectionTests : IClassFixture<CosmosDbFixture>, IDisposable
+    {
+      private readonly HttpClient httpClient;
+
+      private readonly CosmosClient cosmosClient;
+
+      public ConnectionTests(CosmosDbFixture cosmosDbFixture)
+        : this(cosmosDbFixture.Container.HttpClient, cosmosDbFixture.Container.ConnectionString)
+      {
+      }
+
+      private ConnectionTests(HttpClient httpClient, string connectionString)
+      {
+        var cosmosClientOptions = new CosmosClientOptions { ConnectionMode = ConnectionMode.Gateway, HttpClientFactory = () => httpClient };
+        this.httpClient = httpClient;
+        this.cosmosClient = new CosmosClient(connectionString, cosmosClientOptions);
+      }
+
+      [Fact(Skip = "Waiting for a working cosmosdb emulator")]
```

Comment:
Can we add the GitHub issue, why we skip those tests?

Reply:
Good idea, done.